content/apply.js

```diff
@@ -1,41 +1,45 @@
-/* global msg API prefs createStyleInjector */
+/* global API msg */// msg.js
+/* global StyleInjector */
+/* global prefs */
 'use strict';
 
-// Chrome reruns content script when documentElement is replaced.
-// Note, we're checking against a literal `1`, not just `if (truthy)`,
-// because <html id="INJECTED"> is exposed per HTML spec as a global variable and `window.INJECTED`.
+(() => {
+  if (window.INJECTED === 1) return;
 
-// eslint-disable-next-line no-unused-expressions
-self.INJECTED !== 1 && (() => {
-  self.INJECTED = 1;
-
-  let IS_TAB = !chrome.tabs || location.pathname !== '/popup.html';
-  const IS_FRAME = window !== parent;
-  const STYLE_VIA_API = !chrome.app && document instanceof XMLDocument;
-  const styleInjector = createStyleInjector({
+  let hasStyles = false;
+  let isTab = !chrome.tabs || location.pathname !== '/popup.html';
+  const isFrame = window !== parent;
+  const isFrameAboutBlank = isFrame && location.href === 'about:blank';
+  const isUnstylable = !chrome.app && document instanceof XMLDocument;
+  const styleInjector = StyleInjector({
     compare: (a, b) => a.id - b.id,
     onUpdate: onInjectorUpdate,
   });
-  const initializing = init();
+  // dynamic about: and javascript: iframes don't have a URL yet so we'll use their parent
+  const matchUrl = isFrameAboutBlank && tryCatch(() => parent.location.href) || location.href;
+
+  // save it now because chrome.runtime will be unavailable in the orphaned script
+  const orphanEventId = chrome.runtime.id;
+  let isOrphaned;
+  // firefox doesn't orphanize content scripts so the old elements stay
```

Comment:
I think Firefox removes old content scripts before injecting the new ones so old scripts have no chance to cleanup.

Reply:
Yes, AFAICT the comment is telling the same thing: "orphanizing" is leaving the old script hanging around without a parent.